src/Components/Clock.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+type Props = {
+  setHasClock: (hasClock: boolean) => void;
+  clockName: string;
+};
+
+type State = {
+  time: Date;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    time: new Date(),
+  };
+
+  timerTime = 0;
+
+  componentDidMount() {
+    window.addEventListener('contextmenu', this.handleContextMenu);
+    window.removeEventListener('click', this.handleClick);
+
+    this.timerTime = window.setInterval(() => {
+      const newTime = new Date();
+
+      // eslint-disable-next-line no-console
+      console.info(newTime.toUTCString().slice(-12, -4));
```

Comment:
You can create function for time formating, because you used it at least 2 times